File: src/main/java/com/goudourasv/domain/institutions/InstitutionsService.java
package com.goudourasv.domain.institutions;

import com.goudourasv.data.institutions.InstitutionsRepository;
import com.goudourasv.http.institutions.dto.InstitutionCreate;
import com.goudourasv.http.institutions.dto.InstitutionUpdate;

import javax.enterprise.context.ApplicationScoped;
import javax.transaction.Transactional;
import java.util.List;
import java.util.UUID;

@ApplicationScoped
public class InstitutionsService {
    private InstitutionsRepository institutionsRepository;

    public InstitutionsService(InstitutionsRepository institutionsRepository) {
        this.institutionsRepository = institutionsRepository;
    }

    @Transactional
    public List<Institution> getFilteredInstitutions(String country, String city) {
        List<Institution> filteredInstitutions = institutionsRepository.getInstitutions(country, city);
        return filteredInstitutions;
    }


    @Transactional
    public Institution getSpecificInstitution(UUID id) {
        Institution specificInstitution = institutionsRepository.getSpecificInstitution(id);
        return specificInstitution;
    }

    @Transactional
    public Institution createInstitution(InstitutionCreate institutionCreate) {
        Institution institution = institutionsRepository.createInstitution(institutionCreate);
        return institution;
    }

    @Transactional
    public boolean deleteSpecificCourse(UUID id) {
        boolean deleted = institutionsRepository.deleteSpecificInstitution(id);
        return deleted;
    }

    @Transactional
    public Institution replaceInstitution(InstitutionCreate institutionCreate, UUID id) {
        Institution updatedInstitution = institutionsRepository.replaceInstitution(institutionCreate, id);
        return updatedInstitution;
    }

    @Transactional
    public Institution partiallyUpdateInstitution(InstitutionUpdate institutionUpdate, UUID id) {

        Institution updatedInstitution = institutionsRepository.partiallyUpdateInstitution(institutionUpdate, id);
        return updatedInstitution;
    }


}

File: src/main/java/com/goudourasv/domain/institutions/InstitutionsService.kt
package com.goudourasv.domain.institutions

import com.goudourasv.data.institutions.InstitutionsRepository
import com.goudourasv.http.institutions.dto.InstitutionCreate
import com.goudourasv.http.institutions.dto.InstitutionUpdate
import java.util.*
import javax.enterprise.context.ApplicationScoped
import javax.transaction.Transactional

@ApplicationScoped
class InstitutionsService(private val institutionsRepository: InstitutionsRepository) {

    @Transactional
    fun getFilteredInstitutions(country: String?, city: String?): List<Institution> {
        return institutionsRepository.getInstitutions(country, city)
    }

    @Transactional
    fun getSpecificInstitution(id: UUID): Institution? {
        return institutionsRepository.getSpecificInstitution(id)
    }

    @Transactional
    fun createInstitution(institutionCreate: InstitutionCreate): Institution {
        return institutionsRepository.createInstitution(institutionCreate)
    }

    @Transactional
    fun deleteSpecificInstitution(id: UUID): Boolean {
        val deleted = institutionsRepository.deleteSpecificInstitution(id)
        return deleted
    }

    @Transactional
    fun replaceInstitution(id: UUID, institutionCreate: InstitutionCreate): Institution {
        return institutionsRepository.replaceInstitution(institutionCreate, id)
    }

    @Transactional
    fun partiallyUpdateInstitution(id: UUID, institutionUpdate: InstitutionUpdate): Institution {
        return institutionsRepository.partiallyUpdateInstitution(institutionUpdate, id)
    }


}